src/pages/pu-trx/[functionId]/PurchaseTransactionForm.js

```diff
@@ -0,0 +1,1662 @@
+                columnsInDropDown={[
+                  { key: 'reference', value: 'Reference' },
+                  { key: 'name', value: 'Name' },
+                  { key: 'flName', value: 'FL Name' }
+                ]}
+                onChange={(event, newValue) => {
+                  fillVendorData(newValue)
+                }}
+                secondFieldName={'header.vendorName'}
+                onSecondValueChange={(name, value) => {
+                  formik.setFieldValue('header.vendorName', value)
+                }}
+                errorCheck={'header.vendorId'}
+                maxAccess={maxAccess}
+                required
+                readOnly={isPosted}
+                displayFieldWidth={3}
+                editMode={editMode}
+              />
+            </Grid>
+            <Grid item xs={2.4}>
+              <FormControlLabel
+                control={
+                  <Checkbox
+                    name='header.isVattable'
+                    checked={formik.values?.header?.isVattable}
+                    disabled={formik?.values?.items && formik?.values?.items[0]?.itemId}
+                    onChange={formik.handleChange}
+                  />
+                }
+                label={labels.VAT}
+              />
+            </Grid>
+            <Grid item xs={2.4}>
+              <ResourceComboBox
+                endpointId={FinancialRepository.TaxSchedules.qry}
+                name='taxId'
+                label={labels.tax}
+                valueField='recordId'
+                displayField={['name']}
+                readOnly={
+                  !formik.values?.header?.isVattable ||
+                  (formik?.values?.items?.length > 0 && formik?.values?.items[0]?.sku)
+                }
+                values={formik.values.header}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('header.taxId', newValue?.recordId || null)
+                }}
+                error={formik.touched.taxId && Boolean(formik.errors.taxId)}
+                maxAccess={maxAccess}
+              />
+            </Grid>
+            <Grid item xs={2.4}>
+              {metalPriceVisibility && (
+                <CustomNumberField
+                  name='KGmetalPrice'
+                  label={labels.metalPrice}
+                  value={formik.values.header.KGmetalPrice}
+                  readOnly
+                />
+              )}
+            </Grid>
+          </Grid>
+        </Fixed>
+
+        <Grow>
+          <DataGrid
+            onChange={(value, action) => {
+              formik.setFieldValue('items', value)
+              action === 'delete' && setReCal(true)
+            }}
+            value={formik?.values?.items}
+            error={formik.errors.items}
+            name='items'
+            columns={columns}
+            maxAccess={maxAccess}
+            disabled={isPosted || !formik.values.header.vendorId || !formik.values.header.vendorId}
+          />
```

Comment:
![image](https://github.com/user-attachments/assets/010d7b2b-7c8f-4fa8-aedf-c9420e344b64)
just make sure if there is anything that should be mandatory